lib/cog_api/fake/relay_groups.ex

```diff
@@ -38,6 +50,17 @@ defmodule CogApi.Fake.RelayGroups do
   end
 
   def add_relay(_, _, %Endpoint{token: nil}), do: Endpoint.invalid_endpoint
+  def add_relay(%{name: name}, %{relay: relay_name}, %Endpoint{token: _}) do
+    relay = Server.show_by_key(Relay, :name, relay_name)
```

Comment:
As with the delete, this seems to be pretty similar. Is it possible to pull out the duplication into another function that is shared?